lib/trino-parquet/src/main/java/io/trino/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -175,72 +228,65 @@ public static Domain getDomain(
         }
 
         if ((type.equals(BIGINT) || type.equals(TINYINT) || type.equals(SMALLINT) || type.equals(INTEGER)) && (statistics instanceof LongStatistics || statistics instanceof IntStatistics)) {
-            Optional<ParquetIntegerStatistics> parquetIntegerStatistics = toParquetIntegerStatistics(statistics, id, column);
-            if (parquetIntegerStatistics.isEmpty() || isStatisticsOverflow(type, parquetIntegerStatistics.get())) {
+            ParquetLongStatistics parquetIntegerStatistics = toParquetLongStatistics(statistics, id, columnName);
+            if (isStatisticsOverflow(type, parquetIntegerStatistics)) {
                 return Domain.create(ValueSet.all(type), hasNullValue);
             }
-            return createDomain(type, hasNullValue, parquetIntegerStatistics.get());
+            return createDomain(type, id, columnName, hasNullValue, parquetIntegerStatistics, statistics);
         }
 
-        if (type instanceof DecimalType && ((DecimalType) type).getScale() == 0 && (statistics instanceof LongStatistics || statistics instanceof IntStatistics)) {
-            Optional<ParquetIntegerStatistics> parquetIntegerStatistics = toParquetIntegerStatistics(statistics, id, column);
-            if (parquetIntegerStatistics.isEmpty() || isStatisticsOverflow(type, parquetIntegerStatistics.get())) {
-                return Domain.create(ValueSet.all(type), hasNullValue);
-            }
-            return createDomain(type, hasNullValue, parquetIntegerStatistics.get(), statisticsValue -> {
-                if (((DecimalType) type).isShort()) {
-                    return statisticsValue;
+        if (type instanceof DecimalType) {
```

Comment:
Why did the scale check get removed?

Reply:
The scale check isn't required because if it's a short decimal, toParquetLongStatistics will convert it to a long and otherwise the else will handle it.